lib/core/src/events/ListenerBag.ts

```diff
@@ -78,6 +78,34 @@ export class ListenerBag {
     return this.listeners.length !== 0
   }
 
+  /**
+   * Checks if this listener bag instance has at least one of `name` assigned
+   *
+   * @param name The class name of the Listener to be queried
+   * @return Whether this instance has at least one of listener `name`
+   */
+  public hasListener (name: string) : boolean {
+    return this.listeners.some(listener => listener.constructor.name === name)
+  }
+
+  /**
+   * Removes any listeners of `name` from the listener bag instance.
+   *
+   * @param name The class name of the Listener to be removed
+   * @return This instance for chaining
+   */
+  public removeListener (name: string) : ListenerBag {
+    const filteredList: Listener[] = []
+
+    this.listeners.forEach(listener => {
+      listener.constructor.name === name ? null : filteredList.push(listener)
+    })
+
+    this.listeners = filteredList
```

Comment:
I wonder if this can be simplified to

```typescript
this.listeners = this.listeners.filter(l => l.constructor.name !== name)
```

Reply:
Good suggestion - I had worried that because some listeners and (probably) all hooks in `this.listeners` would be anonymous, the condition would always fail for those, but it seems that it doesn't... 👍 - I have updated tests to confirm this...

Reply:
Thanks!